File: dotnet/Middleware.Contracts/Events/AccountLookupRequestedEvent.cs
using NServiceBus;

namespace Middleware.Contracts.Events;

/// <summary>
/// Published by PolicyIssuanceAndLifecycleManagement after compliance clears.
/// CustomerIdentityAndRelationshipManagement subscribes and retrieves or creates the account service record.
/// EDA: PolicyLifecycle announces it needs the account record — it does not command CustomerIdentity.
/// </summary>
public sealed class AccountLookupRequestedEvent : IEvent
{
    public string IssuanceId { get; set; } = string.Empty;
    public string AccountId { get; set; } = string.Empty;
    public DateTimeOffset RequestedAt { get; set; } = DateTimeOffset.UtcNow;
}

File: java/common/src/main/java/com/ais/middleware/common/events/customer/AccountLookupRequestedEvent.java
package com.ais.middleware.common.events.customer;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.OffsetDateTime;

/**
 * Published by PolicyIssuanceAndLifecycleManagement after compliance clears.
 * CustomerIdentityAndRelationshipManagement subscribes and retrieves or creates the account service record.
 * EDA: PolicyLifecycle announces it needs the account record — it does not command CustomerIdentity.
 */
public record AccountLookupRequestedEvent(
        String issuanceId,
        String accountId,
        OffsetDateTime requestedAt
) {
    @JsonCreator
    public AccountLookupRequestedEvent(
            @JsonProperty("issuanceId") String issuanceId,
            @JsonProperty("accountId") String accountId,
            @JsonProperty("requestedAt") OffsetDateTime requestedAt
    ) {
        this.issuanceId = issuanceId;
        this.accountId = accountId;
        this.requestedAt = requestedAt;
    }
}
